src/core/src/randomizer.cpp

```diff
@@ -0,0 +1,210 @@
+/*
+ * Hydrogen
+ * Copyright(c) 2002-2008 by Alex >Comix< Cominu [comix@users.sourceforge.net]
+ * Humanizer
+ * Copyright(c) 2018 by Philipp Müller [thegreatwhiteshark@googlemail.com]
+ *
+ * http://www.hydrogen-music.org
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY, without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ */
+
+#include <hydrogen/randomizer.h>
+#include <cstdlib>
+
+namespace H2Core {
+
+Randomizer* Randomizer::__instance = 0;
+
+// Constructor
+Randomizer::Randomizer() :
+	uniformStateInt64( 4101842887655102017LL ) {
+	__instance = this;
+	// Generate a random seed
+	unsigned long long int initialSeed = std::rand();
```

Comment:
`std::rand()` returns `int` which is narrower than `unsigned long long int`, so all high bits are initialized to 0.

BTW, you might consider using [pseudo random number generation utilities int C++ standard library](https://en.cppreference.com/w/cpp/numeric/random).

Reply:
Yes, indeed. But since it is just the seed for a RNG, which will be initialized ones and (hopefully) never finishes its period, I do not consider this a problem.

Considering the C++ library: I just finished reading the chapter on random numbers in the third edition of Numerical Recipes by Press and the authors claimed that a lot of research did happened in this field and one should not trust the generators implemented in standard libraries, especially not std::rand(). And indeed, all the references in the Pseudo-random number generation library predate the discovery of the now standard statistical tests for checking the quality of RNGs. In addition both the best algorithm described by Press and all the combined or composed RNG he suggested to use are not present in this library. This was actually one of the main reason I started looking at the humanizer.